src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Windows.cs

```diff
@@ -10,14 +12,21 @@ internal static partial class GlobalizationMode
         internal static bool Invariant { get; } = GetInvariantSwitchValue();
 
         internal static bool UseNls { get; } = !Invariant &&
-            (GetSwitchValue("System.Globalization.UseNls", "DOTNET_SYSTEM_GLOBALIZATION_USENLS") ||
-                !LoadIcu());
+            (GetSwitchValue("System.Globalization.UseNls", "DOTNET_SYSTEM_GLOBALIZATION_USENLS", defaultValue: true) || !LoadIcu());
 
         private static bool LoadIcu()
         {
             if (!TryGetAppLocalIcuSwitchValue(out string? icuSuffixAndVersion))
             {
-                return Interop.Globalization.LoadICU() != 0;
+                if (Interop.Globalization.LoadICU() == 0)
+                {
+                    string message = "The configuration System.Globalization.UseNls was set to false and a valid ICU package was not found in the system. " +
+                                "For more information you can go to: https://docs.microsoft.com/en-us/dotnet/standard/globalization-localization/globalization-icu#using-nls-instead-of-icu";
```

Comment:
Remove the "/en-us" portion of this URL. Though I'd instead recommend using a fwlink so that we can retarget it post-release. I just registered https://go.microsoft.com/fwlink/?linkid=2147400 if you want to use it.

Reply:
Great! Thank you!

Reply:
Should this be in a resx or is it too low in stack for that?

Reply:
It is too low in the stack. If we put it in a RESX then we would end in an infinite recursion trying to initialize GlobalizationMode.

Reply:
Fixed.